pkg/compact/retention_test.go

```diff
@@ -18,53 +18,269 @@ import (
 	"github.com/prometheus/tsdb"
 )
 
-func TestTimeBasedRetentionPolicyOnEmptyBucket(t *testing.T) {
-	logger := log.NewNopLogger()
-	bkt := inmem.NewBucket()
-
-	testutil.Ok(t, compact.ApplyDefaultRetentionPolicy(context.TODO(), logger, bkt, 24*time.Hour))
-
-	var (
-		want []string
-		got  []string
-	)
-	testutil.Ok(t, bkt.Iter(context.TODO(), "", func(name string) error {
-		got = append(got, name)
-		return nil
-	}))
-
-	testutil.Equals(t, got, want)
-}
+func TestApplyRetentionPolicyByResolution(t *testing.T) {
+	type testBlock struct {
+		id         string
+		minTime    time.Time
+		maxTime    time.Time
+		resolution compact.ResolutionLevel
+	}
+	type args struct {
+		blocks                []testBlock
+		retentionByResolution map[compact.ResolutionLevel]time.Duration
+	}
+	tests := []struct {
```

Comment:
Can we just do anonymous structs? There is no need to have them around.. maybe except `testBlock`

Then we could just do:
```
for _, tc := range []struct{
  name string
  blocks []testBlock
  retentionByResolution ...
  want []string
  wantErr bool
} {
      {
        ...
      },
      {
      ...
      }
} {
 // test code
}

```
This is kind of general pattern most of go test seems to follow.

Reply:
Hey, sure.
I adjusted to follow this pattern.
Just fyi: the previous code is the pattern generated inside vscode.